Jobba.Core/Implementations/Repositories/RepositoryExpressions.cs

```diff
@@ -8,7 +8,8 @@ namespace Jobba.Core.Implementations.Repositories;
 public static class RepositoryExpressions
 {
     public static readonly Expression<Func<JobEntity, bool>> JobRetryExpression =
-        x => x.Status != JobStatus.Completed && x.Status != JobStatus.Cancelled && !x.IsOutOfRetry;
+        x => (x.Status == JobStatus.Faulted || x.Status == JobStatus.ForceCancelled || x.Status == JobStatus.Unknown) &&
```

Comment:
It was pulling jobs for retry whenever they were in the status of InProgress, Enqueued. I can make it negative again, but I believe we need to include those 2 statuses in the expression if we do

Reply:
Yeah this one is interesting because if this is being used in a distributed environment, like we use it, then if one pod restarts while another pod is running jobs, then it would retry that job. The locks should keep them from running at the same time, but it looks like we would need to update the `IsJobStatusValidAsync` method in the DefaultJobScheduler to check it is was Completed or Cancelled along with Enqueued or InProgress